server/controllers/finance/reports/cashReport/index.js

```diff
@@ -0,0 +1,280 @@
+/**
+ * cashReport Controller
+ *
+ *
+ * This controller is responsible for processing cash report.
+ *
+ * @module finance/cashReport
+ *
+ * @requires lodash
+ * @requires lib/db
+ * @requires lib/ReportManager
+ * @requires lib/errors/BadRequest
+ */
+
+const _ = require('lodash');
+const db = require('../../../../lib/db');
+const ReportManager = require('../../../../lib/ReportManager');
+const BadRequest = require('../../../../lib/errors/BadRequest');
+const accountExtrat = require('../../accounts/extra');
+
+const TEMPLATE_COMBINED = './server/controllers/finance/reports/cashReport/report_combined.handlebars';
+const TEMPLATE_SEPARATED = './server/controllers/finance/reports/cashReport/report_separated.handlebars';
+
+// expose to the API
+exports.document = document;
+
+function getRecordQuery(token, format, openingBalance) {
+  let query;
+
+  if (format === 1) {
+    query =
+      `	
+      SELECT
+        t.trans_id, t.trans_date, t.debit, t.credit, t.description, t.account_id, 
+        a.number, a.label, t.balance, t.cbalance, c.reference AS cashReference, 
+        v.reference AS voucherReference, t.record_uuid
+      FROM
+        (
+          SELECT
+              trans_id, trans_date, debit_equiv AS debit, credit_equiv AS credit, account_id, 
+              description, balance, (@cbal := @cbal + balance) AS cbalance, record_uuid
+          FROM
+          
+            (
+              (
+                    SELECT
+                      p.trans_date, p.debit_equiv, p.credit_equiv, p.trans_id, p.record_uuid,
+                      p.description, p.account_id, (p.debit_equiv - p.credit_equiv) AS balance
+                    FROM
+                      posting_journal AS p
+                    WHERE 
+                      p.account_id= ? AND 
+                      (p.trans_date >= DATE(?) AND p.trans_date <= DATE(?))
+                  )
+                  UNION ALL
+                  (
+                    SELECT
+                      g.trans_date, g.debit_equiv, g.credit_equiv, g.trans_id, g.record_uuid,
+                      g.description, g.account_id, (g.debit_equiv - g.credit_equiv) AS balance
+                    FROM
+                      general_ledger AS g 
+                    WHERE 
+                      g.account_id= ? AND 
+                      (g.trans_date >= DATE(?) AND g.trans_date <= DATE(?))
+                  )
+              ) AS u, (SELECT @cbal := ${openingBalance}
```

Comment:
:+1: Well done!  This is a pretty complicated thing to do, and you've done it well.